gef.py

```diff
@@ -1543,14 +1547,14 @@ def push_context_message(level, message):
     return
 
 
-def show_last_exception():
+def show_last_exception() -> None:
     """Display the last Python exception."""
 
-    def _show_code_line(fname, idx):
+    def _show_code_line(fname: str, idx: int) -> str:
         fname = os.path.expanduser(os.path.expandvars(fname))
         with open(fname, "r") as f:
             __data = f.readlines()
-        return __data[idx - 1] if idx < len(__data) else ""
+        return __data[idx - 1] if 0 <= idx < len(__data) else ""
```

Comment:
if idx is 0, we do `__data[-1]`. Is this correct?

Reply:
TBH i'm not really sure how this change got into this PR.  In General the function is called with idx set to `lineno` of `traceback.FrameSummary` objects and therefore should (probably?) always be positive. So if anything should probably be
```py
return __data[idx - 1] if 0 < idx < len(__data) else ""
```